src/components/intern-app.html

```diff
@@ -135,35 +135,42 @@ <h1>Intern App</h1>
        super();
        this.boundNewUser = this.newUser.bind(this);
        this.boundDeleteUser = this.deleteUser.bind(this);
+       this.boundEditUser = this.editUser.bind(this);
      }
 
      connectedCallback() {
        super.connectedCallback();
        window.addEventListener("newUser", this.boundNewUser);
        window.addEventListener("deleteUser", this.boundDeleteUser);
+       window.addEventListener("editUser", this.boundEditUser);
      }
 
      disconnectedCallback() {
        super.disconnectedCallback();
        window.removeEventListener("newUser", this.boundNewUser);
        window.removeEventListener("deleteUser", this.boundDeleteUser);
+       window.removeEventListener("editUser", this.boundEditUser);
      }
 
-     deleteAll() {
-       if (window.confirm("Are you sure you want to delete all users?")) {
-         this.set('userProfiles', this.deleteUsers());
-       }
+     newUser(event) {
+       const newUserProfile = event.detail;
+       newUserProfile.id = this.localStorageClass.genId();
+       this.pushToArray(this.userProfiles, newUserProfile);
+       this.addUserSelected = false;
+     }
+
+     editUser(event) {
+      this.set('userProfiles', this.localStorageClass.getUsers());
```

Comment:
Since both delete and edit user do the same thing, I'd just make a single function (`updateUsers` maybe) that calls `this.set`.

You could take it a step further and have both your edit and delete component logic dispatch the same message, as well. so it'd eventually look something like

```javascript
connectedCallback() {
    ...
    window.addEventListener('updateUsers', this.boundUpdateUser);
}

...

updateUsers(event) {
    this.set('userProfiles', this.localStorageClass.getUsers());
}
```

and then both edit and delete user logic would dispatch a custom `updateUsers` event, instead of `editUser` or `deleteUser` event

Reply:
I see what you're saying here.  What about the separate calls of .updateUser and .deleteUser here?
<img width="505" alt="Screen Shot 2019-08-29 at 2 32 04 PM" src="https://user-images.githubusercontent.com/29068248/63970488-e7bfe300-ca69-11e9-817b-fa471e9e9b4c.png">

Reply:
so you'd keep all of that, just change the names of the CustomEvent to `updateUsers` (or whatever you want to name the custom event)

Reply:
okay gotcha, I wasn't completely sure which areas to consolidate together there.  Makes sense now.